pika/adapters/tornado_connection.py

```diff
@@ -72,7 +73,10 @@ def _adapter_disconnect(self):
         """Disconnect from the RabbitMQ broker"""
         if self.socket:
             self.ioloop.remove_handler(self.socket.fileno())
-        super(TornadoConnection, self)._adapter_disconnect()
+        try:
+            super(TornadoConnection, self)._adapter_disconnect()
+        except exceptions.AMQPConnectionError as e:
```

Comment:
@awelzel, the reason I proposed the try/finally `BaseConnection._handle_disconnect()` fix above is that it doesn't alter the documented semantics of `on_close_callback` and `on_open_error_callback`, shouldn't negatively impact any other connection classes, and is also deterministic versus "If the connection is neither open, not closed, it \*probably\* was never successfully established".

CC @gmr